File: src/test/java/com/mkayad/busjourney/controllers/MainControllerUnitTest.java
package com.mkayad.busjourney.controllers;

import com.mkayad.busjourney.domain.ArrivalDetail;
import org.junit.jupiter.api.DisplayName;
import org.junit.jupiter.api.Test;
import org.junit.jupiter.api.extension.ExtendWith;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.boot.test.context.SpringBootTest;
import org.springframework.test.context.junit.jupiter.SpringExtension;
import org.springframework.ui.ExtendedModelMap;


import java.util.List;

import static org.junit.jupiter.api.Assertions.assertFalse;
import static org.junit.jupiter.api.Assertions.assertTrue;


@ExtendWith(SpringExtension.class)
@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT)
public class MainControllerUnitTest {

    @Autowired
    private MainController  aMainController;

    @DisplayName("should return a non empty list ")
    @Test
    public void testValidUrlResult() {
        var model = new ExtendedModelMap();
        var response=aMainController.index(model);

        assert(response.equals("index"));

        List<ArrivalDetail> stopGTimes= (List<ArrivalDetail>) model.getAttribute("stopGTimes");
        List<ArrivalDetail> stopHTimes= (List<ArrivalDetail>) model.getAttribute("stopHTimes");

        assertFalse(stopGTimes.isEmpty());
        assertFalse(stopHTimes.isEmpty());
    }




}

File: src/test/kotlin/com/mkayad/busjourney/controllers/MainControllerUnitTest.kt
package com.mkayad.busjourney.controllers

import com.mkayad.busjourney.domain.ArrivalDetail
import org.junit.jupiter.api.Assertions.assertTrue
import org.junit.jupiter.api.DisplayName
import org.junit.jupiter.api.Test
import org.junit.jupiter.api.extension.ExtendWith
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.boot.test.context.SpringBootTest
import org.springframework.test.context.junit.jupiter.SpringExtension
import org.springframework.ui.ExtendedModelMap
import org.springframework.ui.Model


@ExtendWith(SpringExtension::class)
@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT)
class MainControllerUnitTest {

    @Autowired
    lateinit var mainController: MainController

    @DisplayName("should return a non empty list ")
    @Test
    fun testValidUrlResult() {
        val model: Model = ExtendedModelMap()
        val response=mainController.index(model)

        assert(response=="index")

        val stopGTimes:List<ArrivalDetail>?=model.getAttribute("stopGTimes") as? List<ArrivalDetail>
        val stopHTimes:List<ArrivalDetail>?=model.getAttribute("stopHTimes") as? List<ArrivalDetail>

        assertTrue(stopGTimes?.isNotEmpty()!!)
        assertTrue(stopHTimes?.isNotEmpty()!!)
    }




}
